### ISSUE TITLE
Binary cross entropy produces negative results if input is more than 1

### ISSUE THREAD

**Issue by hguandl** [tensorflow/tensorflow#71638]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.6.12

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Mobile device

Linux Ubuntu 22.04

### Python version

3.11.8

### Current behavior?

Output: `tf.Tensor(-1.7812711e-06, shape=(), dtype=float32)`

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

a = tf.Variable([1.0000000597], dtype=tf.float32)
b = tf.constant([1.0], dtype=tf.float32)

loss = tf.keras.losses.binary_crossentropy(a, b)
print(loss)
```

**Comment by tilakrayal** [comment]
@hguandl,
The loss is just a scalar that you are trying to minimize. It's not supposed to be positive! For instance a cosine proximity loss will usually be negative (trying to make proximity as high as possible by minimizing a negative scalar).

https://stackoverflow.com/questions/42264649/keras-binary-crossentropy-has-negative-values/47503934#47503934

Thank you!
